### ISSUE TITLE
Finding out if there is missing alt text in TTW book

### ISSUE THREAD

**Issue by EKaroune** [the-turing-way/the-turing-way#3883]
### Summary
We need to know if images in the TTW book have missing alt text.

### What needs to be done?

- [ ] write some script to find images with missing alt text
- [ ] compile links for these into an issue
- [ ] flag to community so these can be updated

### Who can help?

* infrastructure maintainers

---

### Updates

*Lorem ipsum dolor sit amet, consectetur adipiscing.*

**Comment by JimMadge** [comment]
I've spent a little time looking at some of the MystMD internals. We will probably be moving to MystMD in the future.

The Markdown is parsed into an abstract syntax tree (like you can see [here](https://mystmd.org/sandbox)). With that, it should be fairly easy to iterate over images and figures and make a note of those without an `:alt:` field or with one that is empty.

**Comment by agoose77** [comment]
MyST transforms have access to the `vfile` that represents the current document, and we can there add fatal errors or warnings to the file: https://github.com/jupyter-book/mystmd/blob/d0233001a389ee0360e41f80ea76074bc5bd3516/packages/myst-cli/src/frontmatter.ts#L32

You could build a transform that inspects the AST to identify missing alt text (see these two AST fragments https://mystmd.org/sandbox?tab=ast&myst=IQBbAGEAbAB0AF0AKABjAGEAdAAuAHAAbgBnACkACgAKACEAWwBdACgAYwBhAHQALgBwAG4AZwApAA%3D%3D), e.g. https://next.jupyterbook.org/plugins/directives-and-roles#create-a-transform

**Comment by JimMadge** [comment]
I've made a MySTMD plugin [here](https://github.com/the-turing-way/mystmd-no-alt-text-plugin) that raises warnings for images/figures missing alt text.

It depends on `myst-common` and I haven't found a way to give better feedback about where the problem is (like a Markdown file line number) than the file and image path.

**Comment by JimMadge** [comment]
To take this a step further, we could inspect the quality of the alt-text.

There is a script [here](https://github.com/CivicActions/site-evaluation-tools) which does that and the team are open for feedback and contributions. I saw Mike speaking about this [at FOSDEM](https://fosdem.org/2025/schedule/event/fosdem-2025-4709-alternative-text-for-images-how-bad-are-our-alt-text-anyway-/).

**Comment by JimMadge** [comment]
There is a [PR](https://github.com/jupyter-book/mystmd/pull/1814) to add a warning about missing alt-text in MyST.
